### PR TITLE
PARQUET-1407: Fix binary values from dictionary encoding.

### PR DESCRIPTION
**Pull request by nandorKollar**
Add unit test to demonstrate the issue

### CODE REVIEW

**Comment by nandorKollar** [comment]
@rdblue I added the missing test case. Sorry about the other changes, should I create a separate PR for apache/master instead?

**Comment by rdblue** [comment]
@nandorKollar, yes, please keep PR branches as clean as possible. Also, the best practice is to cherry-pick a contributor's commit and add more commits to it instead of altering contributions. Ideally, also include a ref to that commit in the PR description. We need to keep track of the intent to contribute from the person or organization that owns the original commit.

**Comment by zivanfi** [comment]
@rdblue Please note that this PR is against your private repo that is the basis of the public PR. If you merged this commit into your feature branch that would show up in the public PR as two commits by two authors, but the PR would remain yours and in theory no contribution would be altered.

I suggested this approach to Nandor as at that moment it seemed like a good idea, but it has turned out that the main branch is already ahead of this feature branch. I didn't think of the cherry-picking approach to keep the commit authorship in a new PR, that's a great idea.

**Comment by rdblue** [comment]
@zivanfi, thanks for pointing that out. I didn't notice it was a PR against my branch.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 1]
```diff
@@ -1,14 +1,14 @@
-/* 
+/*
```
Comment (rdblue) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:1]: Please remove whitespace changes.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java [hunk 2]
```diff
@@ -678,6 +656,46 @@ public void testUnionWithSingleNonNullType() throws Exception {
     assertEquals(str("theValue"), nextRecord.get("value"));
   }
 
+  @Test
+  public void testDuplicatedValuesWithDictionary() throws Exception {
```
Comment (rdblue) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:660]: This test fails when the fix is not applied, right?

Comment (nandorKollar) [reply parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWrite.java:660]: Yes, it should fail. Actually it is the example code supplied by the reporter, I just made a unit test from it. The test failed without your fix, and was green with it.

**Comment by nandorKollar** [comment]
@rdblue I created a separate PR (https://github.com/apache/parquet-mr/pull/552) on the public repo, as you suggested.

**Pull request closed without merging**